src/firefly/js/ui/panel/TabPanel.jsx

```diff
@@ -311,8 +209,146 @@ FieldGroupTabs.propTypes = {
+
+
+    const childrenAry = React.Children.toArray(children);
+    // get the headers(Tab)
+    const tabHeaders = childrenAry.map((c, idx) => getHeaderFromTab(c.props, idx, maxTitleWidth));
+
+    const tabListVar = slotProps?.tabList?.variant || 'soft';
+    const activeBg = tabListVar === 'plain' ? 'neutral.softBg' : 'background.surface';
+
+    useEffect(() => {
+        const tbEl = toolbarEl.current;
+        if (tbEl) {
+            const width = tbEl.getBoundingClientRect().width;
+            setMaxTitleWidth( (width/tabHeaders.length) -4 );   // -4 account for margin and padding
+        }
+    }, [toolbarEl?.current?.getBoundingClientRect()?.width]);
+
+    return (
+        <TabList
+            ref={toolbarEl}
+            sx={{
+                flexGrow: 1,
+                overflow: 'auto',
+                scrollSnapType: 'x mandatory',
+                '&::-webkit-scrollbar': { display: 'none' },
+                [`& .${tabClasses.root}`]: {
+                    '&[aria-selected="true"]': {
+                        bgcolor: activeBg,
+                        borderColor: 'divider',
+                        '&::before': {
+                            content: '""',
+                            display: 'block',
+                            position: 'absolute',
+                            height: 2,
+                            bottom: -2,
+                            left: 0,
+                            right: 0,
+                            bgcolor: activeBg,
+                        },
+                    },
+                },
+            }}
+            {...{variant:tabListVar, ...slotProps?.tabList}}
+        >
+            {tabHeaders}
+        </TabList>
+    );
+}
+
+function getHeaderFromTab({name, value, label, startDecorator, removable, onTabRemove, ...rest}, idx, maxTitleWidth) {
+    const {selected, onSelect, style, id, ...joyTabProps} = rest;     // deprecated; filtered out.
+
+    // to support deprecated props
+    label ??= name;
+    value ??= id ?? idx;
+
+
+    if (isString(label)) {
+        label = <Box title={label} sx={{whiteSpace:'nowrap', overflow:'hidden', textOverflow:'ellipsis'}}>{label}</Box>;
+    }
+
+    return (
+        <JoyTab key={idx} component='div' {...{value, indicatorPlacement:'top', ...joyTabProps}}
+                sx={{
+                    flex: 'none', scrollSnapAlign: 'start',
+                    maxWidth: maxTitleWidth,
+                    minWidth: 100
+                }}
+        >
+            {startDecorator && (
+                <ListItemDecorator>
+                    {startDecorator}
+                </ListItemDecorator>
+            )}
+            {label}
+            {removable &&
+                <Chip variant='soft'
+                      onClick={(e) => {
+                          onTabRemove && onTabRemove(name);
+                          e.stopPropagation && e.stopPropagation();
+                }}>x</Chip>
```

Comment:
I just read out Chip. There is a `ChipDelete`, maybe we should start using that.